### PR TITLE
Use official rust image and clean up Dockerfiles

### PR DESCRIPTION
**Pull request by 6543**
- Prepair buildx to compile for both ARM and x86 architectures.
- Make the Dockerfile buildable on both ARM and x86 architectures.

Note: Due to https://github.com/briansmith/ring/issues/1414, we cannot cross-compile during the build step. So we either emulate the build step but that will increase build time significant, or we still have to use dedicated agents as it is proposed now

### CODE REVIEW

**Comment by 6543** [comment]
works-> https://woodpecker.join-lemmy.org/LemmyNet/lemmy/pipeline/110/4 :tada: 

if you wana keep the docker test build or not ... is up to you, but It is time consuming so I would remove it

**Comment by 6543** [comment]
next test: https://woodpecker.join-lemmy.org/LemmyNet/lemmy/pipeline/114/3

will push a revert commit to the last one to restore the pipeline to what it should alter too ...

**Comment by 6543** [comment]
@dessalines you can already review the Dockerfile(s) and decide if we should delete the dockerfile.arm one now?

**Comment by 6543** [comment]
WIP: cross compile ssl

**Comment by 6543** [comment]
move cross compile into https://github.com/LemmyNet/lemmy/pull/2806 as it's blocked by dep "ring" :/

**Comment by 6543** [comment]
@dessalines @Nutomic ready to review

**Review by dessalines — APPROVED** [review]
Wow thank you for doing this! 

Is the `Dockerfile.debian` just for your own setup?

File: docker/Dockerfile.multiarch [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+FROM alpine:3 as lemmy
+
+# Install libpq for Postgres
+RUN apk add --no-cache ca-certificates libpq
+
+# Copy resources
+COPY --from=builder /app/lemmy_server /app/lemmy
+
+EXPOSE 8536
+CMD ["/app/lemmy"]
```
Comment (Nutomic) [inline docker/Dockerfile.multiarch:55]: Should probably remove this file as cross-compile isnt working. And based on the age of the issue it wont get fixed anytime soon :(

File: docker/Dockerfile.debian [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# Create a new user and group
+RUN addgroup --gid 1000 lemmy && \
+    useradd --no-create-home --shell /bin/sh --uid 1000 --gid 1000 lemmy
+
+# Copy resources and set permissions
+COPY --chmod=755 --from=builder /app/lemmy_server /app/lemmy
+
+USER lemmy
+EXPOSE 8536
+CMD ["/app/lemmy"]
```
Comment (Nutomic) [inline docker/Dockerfile.debian:40]: Is there any reason for adding a Debian dockerfile? I dont think we want to maintain another one.

File: docker/Dockerfile.multiarch (hunk 1 above)
Comment (6543) [reply docker/Dockerfile.multiarch:55]: This pull does not address cross compilation!

Its moved in a extra pull

File: docker/Dockerfile.debian (hunk 2 above)
Comment (6543) [reply docker/Dockerfile.debian:40]: I just did not throw away the arm one ... and as it's not specific to arm anymore ... i did rename it to debian... as that's the base

I'm total fine with just deleting it

Comment (6543) [reply docker/Dockerfile.debian:40]: @dessalines ☝️

Comment (dessalines) [reply docker/Dockerfile.debian:40]: Ya I agree with deleting it. The less dockerfiles to maintain, the better.

**Comment by 6543** [comment]
Deleted :)

File: docker/Dockerfile.multiarch (hunk 1 above)
Comment (Nutomic) [reply docker/Dockerfile.multiarch:55]: Looks like this was added to the main branch anyway.

Comment (6543) [reply docker/Dockerfile.multiarch:55]: the multiarch docker image is just ment to be build with buildx ... and the cross-compiline is just an optional feature ... for now I would recomend to just use it as single source of truth and have different agents that target this dockerfile

the old Dockerfile is just for normal devs who wana build the image but dont have [buildx](https://docs.docker.com/engine/reference/commandline/buildx/) installed

Comment (Nutomic) [reply docker/Dockerfile.multiarch:55]: In that case I would prefer to remove the old dockerfile, and list buildx as a requirements. cc @dessalines

Comment (dessalines) [reply docker/Dockerfile.multiarch:55]: Yep, I'm def fine with removing the old `Dockerfile`

Comment (6543) [reply docker/Dockerfile.multiarch:55]: Well the "old" is the same as the new one ... just without some fancy if statements and args on FROM 😆. so you can make the multiarch to a normal one again at any time

Comment (6543) [reply docker/Dockerfile.multiarch:55]: #2818

Comment (6543) [reply docker/Dockerfile.multiarch:55]: -> https://github.com/LemmyNet/lemmy/pull/2818#issuecomment-1519256243

**Pull request merged**